music21/romanText/writeRoman.py

```diff
@@ -255,27 +273,36 @@ def prepSequentialListOfLines(self):
                     msg = f'further time signature change(s) unprocessed: {unprocessedTSs}'
                     self.combinedList.append(f'Note: {msg}')
 
-            # RomanNumerals
-            measureString = ''  # Clear for each measure
+            # Initialise measure
+            thisMeasureNumber = thisMeasure.measureNumberWithSuffix()  # now str
+            measureString = 'm' + thisMeasureNumber
 
-            rnsThisMeasure = thisMeasure.getElementsByClass(roman.RomanNumeral)
+            # Start repeat
+            if thisMeasure.leftBarline:
+                if 'Repeat' in thisMeasure.leftBarline.classes:
+                    measureString += ' ||:'  # immediately after measure number
 
+            # Roman Numerals
+            rnsThisMeasure = thisMeasure.getElementsByClass(roman.RomanNumeral)
             for rn in rnsThisMeasure:
-                if rn.tie is None or rn.tie.type == 'start':  # Ignore tied to Roman numerals
+                if rn.tie is None or rn.tie.type == 'start':  # Ignore tied-to Roman numerals
                     chordString = self.getChordString(rn)
-                    measureString = rnString(measureNumber=thisMeasure.measureNumber,  # Suffix?
-                                             beat=rn.beat,
-                                             chordString=chordString,
-                                             inString=measureString,  # Creating update
-                                             )
+                    if rn.beat != 1:  # b1 not needed
+                        measureString += f' b{intBeat(rn.beat)}'
+                    measureString += f' {chordString}'
```

Comment:
please explain the change here -- we're removing tested and testable code for a single RN and replacing it with nested code that is hard to test.  -1